### ISSUE TITLE
Support user-data-url in addition to user-data metadata (in GCE and other environments?)

### ISSUE THREAD

**Issue by alex-mohr** [coreos/coreos-overlay#1073]
Google Compute Engine has  a 32kB limit for metadata values.  The instructions for using CoreOS on GCE include passing cloud-init config via such a metadata value: https://coreos.com/docs/running-coreos/cloud-providers/google-compute-engine/

One user (@enach?) reported that only allowed 17 docker containers to be specified before the 32kB limit is hit: https://groups.google.com/forum/#!topic/gce-discussion/tfvecY061pY

Please consider supporting a "user-data-url" key in addition to "user-data", where that key's value is a url for an arbitrarily-large file, similar to "Using a startup script from Google Cloud Storage" at https://cloud.google.com/compute/docs/startupscript#googlecloudstorage.

I think relevant implementation is: https://github.com/coreos/coreos-overlay/blob/master/coreos-base/oem-gce/files/gce-coreos-cloudinit

**Comment by alex-mohr** [comment]
Sorry, initial user report was probably @enachb?

**Comment by enachb** [comment]
Yes, we are seeing this issue on GCE. A "user-data-url" would indeed give us a bit more flexibility.
